sharktank/sharktank/utils/tree.py

```diff
@@ -6,6 +6,7 @@
 
 from typing import Dict, Any, Callable, Mapping, Iterable, Sequence, Union
 from collections.abc import Mapping as MappingABC, Iterable as IterableABC
+import functools
```

Comment:
You'll want someone else to review this file, since I haven't worked with the tree functionality before.

Reply:
I don't think anyone else is. This is some general functionality for tree manipulation. If it is not clear I can add more documentation.